server/app.js

```diff
@@ -1,15 +1,152 @@
+  console.log("Middleware running now!");
+  console.log(`${method} for the path "${url}" of db "exercise"`);
+  next();
+});
+
+app.get("/course", (req, res) => {
+  Course.find()
+    .then((course) => {
+      res.json(course);
+      res.status(200);
+      console.log(course);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({ error: `Internal Server Error ${error}` });
+    });
+});
+
+app.get("/course/:courseId", (req, res) => {
+  const { courseId } = req.params;
+
+  Course.findById(courseId)
+    .then((post) => {
+      res.status(200);
+      res.json(post);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server error ${error}`,
+      });
+    });
+});
+
+app.get("/students", (req, res) => {
+  Students.find()
+    .then((course) => {
+      res.status(200);
+      res.json(course);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json(error);
+    });
+});
+
+app.get("/students/:studentsId", (req, res) => {
+  const { studentsId } = req.params;
+  Students.findById(studentsId)
+    .then((post) => {
+      res.status(200);
+      res.json(post);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server error ${error}`,
+      });
+    });
+});
+
+app.get("/course/:id/students/:id", (req, res) => {
+  Course.findById(req.params.id)
+    .then((post) => {
+      res.status(200);
+      res.json(post);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json({
+        error: `Internal Server error ${error}`,
+      });
+    });
+});
 
-app.get("/", (req, res) => {
-  res.json({
-    "/courses": "nothing yet",
+app.post("/course", (req, res) => {
+  Course.create({
+    name: req.body.name,
+    type: req.body.type,
+    location: req.body.location,
+  })
+    .then((course) => {
+      res.status(201);
+      res.status(course);
+      console.log(course);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json(error);
+    });
+});
+
+app.post("/students", (req, res) => {
+  const { courseId } = req.params;
+  Students.create({
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    location: req.body.location,
+    course: req.body.course,
+  })
+    .then((student) => {
+      res.status(201);
+      res.status(student);
+      console.log(student);
+    })
+    .catch((error) => {
+      res.status(500);
+      res.json(error);
+    });
+});
+
+app.delete("/courses/:coursesId", (req, res) => {
+  const { courseId } = req.params;
+  Course.findByIdAndDelete(courseId).then(() => {
+    res.status(204);
+    res.json("course eleted");
   });
 });
 
-app.listen(4000, () => {
-  console.log("Listening on http://localhost:4000");
+app.delete("students/:studentsId", (req, res) => {
+  const { studentsId } = req.params;
+  Students.findByIdAndDelete(studentsId).then(() => {
+    res.status(204);
+    res.json("student deleted");
+  });
```

Comment:
I would've expected the deleted entry to be returned instead of just a message. Just like the courses one: https://github.com/marwinburesch/api-from-scratch/wiki/DELETE

Also, the `json()` method expects a JSON to be passed as argument, You're just passing a string :)
http://expressjs.com/en/api.html